src/sync/installation.ts

```diff
@@ -372,12 +387,14 @@ export const processInstallation =
 
 				await subscription.update({ syncStatus: "FAILED" });
 
-				statsd.increment(metricSyncStatus.failed);
 				logger.warn(err, `Sync failed: installationId=${installationId}`);
 
 				job.sentry.setExtra("Installation FAILED", JSON.stringify(err, null, 2));
 				job.sentry.captureException(err);
 
+				const tags = { errorMsg: JSON.stringify(err, null, 2) };
+				statsd.increment(metricSyncStatus.failed, tags);
```

Comment:
So the problem with this approach is that we are risking to explode the metric cardinality. 

We don't know the shape of `err` at this point and it can simply include things like "Installation ID" which will change from customer to customer. We potentially have here a set of unbounded tag variations which we should avoid.

If we want to have a cause in metrics, we should map the error to some kind of Enum, where we strictly control possible variations. And then use this enum as a tag value. (Or, In Java we could use an exception class name hence we don't have many possible classes of errors, but never an exception message)

Reply:
So, until we have enough insight into what the errors are and can map them accordingly, would you recommend just sending logs to splunk and capturing the error with sentry? I can make a follow-up issue in Jira to map errors and add an errorMsg chart after we've collected enough data.

Reply:
Ah, I just commented on that in a separate comment :).

I think it's fine to explode cardinality here, because this error (hopefully) doesn't happen that often. 

I agree that ideally, we would map the errors to an enum, but that's only possible if we know the arrors, as @rachellerathbone pointed out in the description of this PR. 

We could maybe just cut the error message off after 100 characters or so, for now? Then see where we land with the metrics, and then at some point classify them properly and change the metric again.

Reply:
I don't think cutting the message is an option hence even in 100 characters there is N^100 variations of the string. (We never know what will be there and identifiers can be in this first 100 characters)

I've send a link explaining the cardinality issue to Slack

Reply:
So I think the purpose of SFX metrics is a bit different. They can indicate that we have a failures spike. We use it to detect anomalies or watch trends. If we see in SFX that something is not right, we go to Splunk and query the exact errors. Bucketing should be enough. If we see too many errors in "Other" bucket we go to Splunk an then investigating if we should make a new bucket, fix error, etc.

Reply:
Just removed the tags. We can use splunk/sentry for error messages.